plugins/browser-plugin-focalmeter/src/index.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2022 Snowplow Analytics Ltd, 2010 Anthon Pang
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import {
+  attemptGetLocalStorage,
+  BrowserPlugin,
+  hasLocalStorage,
+  BrowserTracker,
+  attemptWriteLocalStorage,
+} from '@snowplow/browser-tracker-core';
+import { Logger, LOG_LEVEL, Payload } from '@snowplow/tracker-core';
+
+/**
+ * The FocalMeter Plugin
+ *
+ * The plugin sends requests with the domain user ID to a Kantar endpoint used with the FocalMeter system.
+ * A request is made when the first event with a new user ID is tracked.
+ *
+ * @param kantarEndpoint URL of the Kantar endpoint to send the requests to (including protocol)
+ * @param useLocalStorage Whether to store information about the last submitted user ID in local storage to prevent sending it again on next load
+ * @param logLevel Log message level
+ * @returns The plugin
+ */
+export function FocalMeterPlugin(
+  kantarEndpoint: string,
```

Comment:
❓ 
Can we make this accept an options object instead of positional arguments ? 
IMO it will help with the maintenance of the plugin and adding new things if needed _(probably this might not need any updates but I would consider it a good practise)_. WDYT

Reply:
Sounds like a good idea! I'll change it.

Reply:
Added the `FocalMeterConfiguration` type to wrap the options. Also added an `enableFocalMeterIntegration` function to pass the configuration object since that is how it is done in other plugins (e.g., form tracking). Also passing the options when creating the plugin would not be possible in the JS tag interface which does not have a way to do that, so using the enable function seems better.